packages/react-components/src/components/utils/Datetime.ts

```diff
@@ -7,50 +7,26 @@ import { MessageThreadStrings } from '../MessageThread';
  * @private
  */
 export const formatTimeForChatMessage = (messageDate: Date): string => {
-  let hours = messageDate.getHours();
-  let minutes = messageDate.getMinutes().toString();
-  const isAm = hours < 12;
-  if (hours > 12) {
-    hours = hours - 12;
-  }
-  if (hours === 0) {
-    hours = 12;
-  }
-  if (minutes.length < 2) {
-    minutes = '0' + minutes;
-  }
-  return hours.toString() + ':' + minutes + ' ' + (isAm ? 'a.m.' : 'p.m.');
+  return messageDate.toLocaleTimeString([], { hour: 'numeric', minute: '2-digit' });
 };
 
 /**
  * @private
  */
 export const formatDateForChatMessage = (messageDate: Date): string => {
-  const year = messageDate.getFullYear().toString();
-  let month = (messageDate.getMonth() + 1).toString();
-  let day = messageDate.getDate().toString();
-
-  if (month.length === 1) {
-    month = '0' + month;
-  }
-  if (day.length === 1) {
-    day = '0' + day;
-  }
-
-  return year + '-' + month + '-' + day;
+  return messageDate.toLocaleDateString();
 };
 
 /**
- * Given a message date object in ISO8601 and a current date object, generates a user friendly timestamp text like the
- * following:
- *
- * 1:30 p.m.
- * Yesterday 1:30 p.m.
- * Monday 1:30 p.m.
- * 2021-01-10 1:30 p.m.
+ * Given a message date object in ISO8601 and a current date object, generates a user friendly timestamp text
+ * using the system locale.
+ * <time in locale format>.
```

Comment:
can be added for all examples: `e.g. 1:30pm`

Reply:
Now we have a variable format, it's not really possible to define what will happen. 24hr vs 12hr, different date separators for different locales etc.
I've added templated examples but really can't go into every possibility :)